### ISSUE TITLE
FTP access ?

### ISSUE THREAD

**Issue by tr1p0p** [YunoHost-Apps/grav_ynh#107]
Like it's been said A LOTS on the grav documentations. User need a proper FTP access to /var/www/grav folders to easily create folders and modular pages, for example. Things admin plugin seem to don't handle well (?!?)

So... Why grav (whith a star and said that a well managed and good package) don't send any info on the admin mail in the creation ? Why the user don't have FTP access to /var/www/grav (i figure it's because app is a user, but.. it's not very practical at all then).

Because if it's AGAIN a app that need admin SSH and sudo to just... create folders put files and edit them, it's like not usable at all then. It's not a "easy CMS" if it retire this easy access to grav folder, and so, the website.

For some reasons, waht i said here didn't found anywhere on the README, didn't have a issues or some steps to just... access the grav folders.

Permissions should protect user, yeah, but not take away possibility of an web app

**Comment by tituspijean** [comment]
Let me rephrase for you.

- [ ] Add `sftp.app` permission
- [ ] Add `ssh.app` permission

**Comment by tituspijean** [comment]
@tr1p0p there. Upgrade with the following command and let me know if you are satisfied.

`yunohost app upgrade grav -u https://github.com/YunoHost-Apps/grav_ynh/tree/ssh-ftp -F`

**Comment by tr1p0p** [comment]
Ok well i update my grav with this but it don't seem to work for the credentials i think my app use (i mean it's `grav`, right ?) get acces denied on filezilla or ssh terminal.

I activate ssh sftp acces in app's config panel, try to change password and all but don't seem to work.

I mean to acces my app i need to log in with `grav` and my password i put on app panel, right ?

**Comment by tituspijean** [comment]
I only populated it with a random password. You should be able to change it and it should work directly though. Can you describe what happens if that does not work?

I'll update the README to better explain what to do.

**Comment by tr1p0p** [comment]
Well, i put like a passphrase, save, no error message, but the pass i put just didn't work. And go back to app settings seem to put a new FTP password (that, then work)

**Comment by tituspijean** [comment]
So you mean you had to change the password twice for it to work?

**Comment by Qprq** [comment]
> @tr1p0p there. Upgrade with the following command, enable SSH/SFTP access in the app's config panel, and let me know if you are satisfied.
> 
> `yunohost app upgrade grav -u https://github.com/YunoHost-Apps/grav_ynh/tree/ssh-ftp -F`

I am satisfied!

I followed your directions, and everything works as intended (sftp and ssh).

Thank you for this valuable addition to this app.

**Comment by cloo** [comment]
I still cannot use Grav via FTP, and I'm confused by the system of users.

If I understand correctly, there are at least three levels of users here: 
1. Yunohost user (my SSO), which somehow can be used as login to Grav
2. Grav user, "admin", which was created via Yunohost. It is magically connected to my Yunohost SSO user (which has a different name), and uses its password.
3. Linux user named "grav", who owns all files and folders, has no visible password, and as you wrote here, it should be used for ftp access.

My SSO and "admin" password does not work for "grav" account when logging via FTP. 

I can strangely login via FTP using "admin", but I cannot edit files:

![220606-010701](https://user-images.githubusercontent.com/651760/172072667-51b31abb-adaa-497b-8e5a-702213b97584.png)


The upgrade command which was provided here worked with ambiguous response, with both failure and success lines, not sure what to think :)
It also suggested to upgrade via web ui, which I did. After that tried this command again, and it worked exactly the same way. Didn't help with ftp though.

![220606-004329](https://user-images.githubusercontent.com/651760/172072919-40adf6fe-e47f-455b-a19a-4d63a4494b56.png)

So.. I'd be thankful for some help for ftp login..

**Comment by Qprq** [comment]
@cloo the standard version of the app hs no FTP access. If you install the _ssh-ftp_ branch as described here, you will be able to activate such access by going to the config page of the app in yunohost admin (🏠/Applications/grav + clic on the yellow button). You will setup a password to use with the _grav_ user to connect with ssh or sftp (port 22).

**Comment by cloo** [comment]
Sorry, it still does not work. 

- Upgraded to Grav 1.7.34~ynh1
- Enabled SSH via Yunohost admin UI
- Can log in via SSH, but cannot edit content. Looks like read-only. Same result when loggin via my Yunohost SSO user and GRAV admin user.

**Comment by tituspijean** [comment]
Sorry, but I cannot replicate your issue.

1. Grav 1.7.34~ynh1
2. Enabled SSH/SFTP in the app config panel and set my own password: 
![image](https://user-images.githubusercontent.com/8769166/184497624-1618eb6b-bc36-494b-8ef5-e32f1d9613d7.png)

3. Logged in with Filezilla with user `grav` and set password, and was able to edit files: 
![image](https://user-images.githubusercontent.com/8769166/184497668-cd2afa62-59b0-4c25-9aaa-f5456ba0aa81.png)

Do not attempt to use any other user than `grav` (or `grav__2`, etc. if you have multiple Grav instances). It will *not* work. If your issue persists, please open a dedicated issue or a forum thread with as many logs and screenshots as possible to understand what you are doing. Especially, share the output of `ls -la /var/www/grav`.

**Comment by cloo** [comment]
I never thought about using username "grav" :) - I have "admin" and another for yunohost ;) 
It would be good to write it in the docs, or even better, in UI where you enable ssh.

**Comment by cloo** [comment]
@tituspijean, just a separate public thanks for doing this. Now we have fully usable Grav on Yunohost!

**Comment by tituspijean** [comment]
> It would be good to write it in the docs, or even better, in UI where you enable ssh.

My bad here... it was so obvious for me that writing it out went over my head.
